From b04c4d29c537cf76f3e5b7047b97433f64b29e27 Mon Sep 17 00:00:00 2001
Message-Id: <patch-5.6-b04c4d29c53-20210329T161606Z-avarab@gmail.com>
In-Reply-To: <cover-0.6-00000000000-20210329T161606Z-avarab@gmail.com>
References: <20210307132001.7485-1-avarab@gmail.com>
	<cover-0.6-00000000000-20210329T161606Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Mon, 29 Mar 2021 17:53:27 +0200
Subject: [PATCH v2 5/5] Makefile: don't "rm configure" before generating it
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Remove a "rm configure" before we make a new "configure" file via a
"configure+" temporary file. This was partially made redundant via the
new ".DELETE_ON_ERROR" target.

"Partially" because if we die partway we'll remove the "configure"
now, but as before we'll leave the "configure.ac+" in place. Ideally
we'd have another target for the "configure.ac+" so this would all
work properly, but due to the logic added in 122650457a6 (build: do
not automatically reconfigure unless configure.ac changed, 2013-01-02)
that's non-trivial to untangle.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 Makefile | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 573bce20357..ce76c476a3c 100644
--- a/Makefile
+++ b/Makefile
@@ -2358,8 +2358,7 @@ $(SCRIPT_PYTHON_GEN): % : unimplemented.sh
 	mv $@+ $@
 endif # NO_PYTHON
 
-CONFIGURE_RECIPE = $(RM) configure configure.ac+ && \
-		   sed -e 's/@@GIT_VERSION@@/$(GIT_VERSION)/g' \
+CONFIGURE_RECIPE = sed -e 's/@@GIT_VERSION@@/$(GIT_VERSION)/g' \
 			configure.ac >configure.ac+ && \
 		   autoconf -o configure configure.ac+ && \
 		   $(RM) configure.ac+
-- 
2.31.1.461.gd47399f6574

